unittests/cpp/common/tests/ParamWithValue_test/ParamWithValue_test.cpp

```diff
@@ -270,3 +277,109 @@ TEST_F(ParamWithValueTest, ParamInfoToProto_Error) {
+    catena::common::ParamWithValue<catena::common::EmptyValue> childParam(catena::common::emptyValue, childParamDescriptor, dm, false);
+    
+    catena::common::ParamDescriptor child2ParamDescriptor(st2138::ParamType::EMPTY, {}, {{"en", "EQ List2"}}, "", "", false, "f3", "", nullptr, false, false, dm, 0, 0, 2, false, &childParamDescriptor);
+    catena::common::ParamWithValue<catena::common::EmptyValue> child2Param(catena::common::emptyValue, child2ParamDescriptor, dm, false);
+    
+    // Expect authorization checks for each level
+    EXPECT_CALL(authz_, readAuthz(testing::A<const IParamDescriptor&>())).Times(3).WillRepeatedly(testing::Return(true));
+    
+    // Test: Navigate through multi-level path
+    Path path = Path("/f2/f3");
+    EXPECT_TRUE(parentParam.getParam(path, authz_, rc_));
+    EXPECT_EQ(rc_.status, catena::StatusCode::OK);
+ }
+/*
+ * TEST 17 - Testing getParam when index is empty
+ * This test verifies the behavior when index is empty in the getParam method.
+ * When index is empty, the method should not recurse further and should check authorization.
+ */
+TEST_F(ParamWithValueTest, NumericIndexOnStructParameterFails) {
+    // Create a parent parameter with a child parameter
+    Device dm;
+    
+    catena::common::ParamDescriptor parentDescriptor(st2138::ParamType::STRUCT, {}, {{"en", "Audio Channel"}}, "", "", false, "f1", "", nullptr, false, false, dm, 0, 0, 2, false, nullptr);
+    catena::common::ParamWithValue<catena::common::EmptyValue> parentParam(catena::common::emptyValue, parentDescriptor, dm, false);
+
+    // Authorization should be checked even for invalid access attempts
+    EXPECT_CALL(authz_, readAuthz(testing::A<const IParamDescriptor&>())).Times(1).WillRepeatedly(testing::Return(true));
+    
+    // Test: Try to access struct parameter by numeric index "/0" - this should fail
+    Path path = Path("/0");
+    EXPECT_FALSE(parentParam.getParam(path, authz_, rc_));
+    EXPECT_EQ(rc_.status, catena::StatusCode::INVALID_ARGUMENT);
+}
+/*
+ * TEST 18 - Testing getParam's authorization failure.
+ * Verifies that getParam returns PERMISSION_DENIED when authorization check fails
+ */
+TEST_F(ParamWithValueTest, UnauthorizedParameterAccessDenied) {
+    // Create a parent parameter with a child parameter
+    Device dm;
+    
+    catena::common::ParamDescriptor parentDescriptor(st2138::ParamType::STRUCT, {}, {{"en", "Audio Channel"}}, "", "", false, "f1", "", nullptr, false, false, dm, 0, 0, 2, false, nullptr);
+    catena::common::ParamWithValue<catena::common::EmptyValue> parentParam(catena::common::emptyValue, parentDescriptor, dm, false);
+
+    catena::common::ParamDescriptor childParamDescriptor(st2138::ParamType::EMPTY, {}, {{"en", "EQ List"}}, "", "", false, "f2", "", nullptr, false, false, dm, 0, 0, 2, false, &parentDescriptor);
+    
+    // Passes Authz twice then fails the rest
```

Comment:
```suggestion
    // Passes Authz once then fails the rest
```
it only passes once.